readme: rewrite the readme for a better help documentation (#543)
bump-sh/cli: README.md

## Installation

Bump CLI is a node package, currently distributed via NPM which means you need to have Node v14+ interpreter installed on your computer or CI servers.

_If you are looking to use Bump in a continuous integration environment you might be interested by [our Github Action](https://github.com/marketplace/actions/api-documentation-on-bump)._

> You can download a standalone package directly from the latest

> > Github release assets if you don’t use Node. We plan to distribute

Author: We should probably remove this sentence indeed as it's been 2yrs since it's there :grimacing:

> > Github release assets if you don’t use Node. We plan to distribute
> > universal binaries to common package managers soon. Please check our
> > installation methods for updates.

Author: ```suggestion
> Github release assets if you don’t use Node.
```

{: .info}

### Global installation

To install it globally, run the following command with NPM

```sh-session
npm install -g bump-cli
```

Or, with Yarn via

```sh-session
yarn global add bump-cli
```

### Add Bump.sh to your node project

As our CLI is a node package, you can easily embed it to your project by adding the package to your `package.json` file, either with NPM

```sh-session
npm install --save-dev bump-cli
```

Or with Yarn via

```sh-session
yarn add --dev bump-cli
```

You can then use any Bump commands with `npx` (same as `npm exec`)

```sh-session
npx bump --help
```

### How should I do if I'm not using npm ?

> Unfortunately, at the moment we only support the Node environment. However we plan to distribute universal binaries in the most common package managers very soon. In the meantime, you can download a standalone package directly from the [latest Github release](https://github.com/bump-sh/cli/releases) assets or you can push your documentation using [our API](https://developers.bump.sh/) (advanced usage only).

Author: ```suggestion
Unfortunately, at the moment we only support the Node environment. However, you can download a standalone package directly from the [latest Github release](https://github.com/bump-sh/cli/releases) assets which you can run as a standalone binary. Or you can push your documentation using [our API](https://developers.bump.sh/) (advanced usage only).
```

  preview  Create a documentation preview from the given file or URL.
```

 You can also get some help anytime by adding `--help` to any command. Example: `bump deploy --help`.

## Prepare your Bump.sh account

While some commands don't need any API token (`preview` or `diff`) you will need an access key if you want to interact with your Bump.sh documentation.

Head to your Documentation Settings in the “CI deployment” section, or your Account or Organization Settings in the “API keys” section to fetch a personnal token for later usage.

## Commands

* [`bump deploy [FILE]`](#bump-deploy-file)
* [`bump diff [FILE]`](#bump-diff-file)
* [`bump preview [FILE]`](#bump-preview-file)

### `bump deploy [FILE]`

> When you update your API, you want its documentation to be live for your API users. This is what the deploy command is for. Publish a new API document with this command, and Bump.sh will analyse your API structure and generate a changelog item if the API structure has changed. It will also update your latest published document with the deployed file.

Author: You are right, let's remove the second sentence

  Response modified: 200
    [Breaking] Body attribute modified: energie
```

> > You can create as many diffs as you like without being authenticated. This is a **free and unlimited service** provided you have a fair use of the service.

Reviewer: ```suggestion
> You can create as many diffs as you like without being authenticated. This is a **free and unlimited service** provided as long as you use the service fairly.
```

{: .info}

_Note: You can also test this feature in our dedicated web application at <https://api-diff.io/>._

#### Authenticated diffs related to your Bump.sh documentation

From an existing Bump.sh documentation, the `diff` command will retrieve a comparison changelog between your latest published documentation and the given file or URL:

```sh-session
bump diff path/to/your/file.yml --doc my-documentation --token $DOC_TOKEN
```

If you want to compare two unpublished versions of your API document, the `diff` command can retrieve a comparison changelog between two given file or URL, “as simple as `git diff`”:

```sh-session
bump diff path/to/your/file.yml path/to/your/next-file.yml --doc my-documentation --token $DOC_TOKEN
```

Please check `bump diff --help` for full usage details.

### `bump preview [FILE]`

> When writing a documentation, you might want to preview how it renders on Bump.sh. This is exactly the goal of the `preview` command: it will create a temporary documentation with a unique URL, which will be available for a short period of time.

Reviewer: ```suggestion
When writing a documentation, you might want to preview how it renders on Bump.sh. This is precisely the goal of the `preview` command: it will create a temporary documentation with a unique URL, which will be available for a short period.
```

Usage from a local OpenAPI or AsyncAPI file

```shell
bump preview path/to/file.json
```

You can also preview a file available from a URL

```shell
bump preview https://developers.bump.sh/source.yaml
```

#### Live preview

By using the `--live` flag you can stay focused on API design (OpenAPI or AsyncAPI file) while seeing a continuously updated preview each time you save your API document.

- Launch the live preview command in your terminal

```shell
bump preview --live --open openapi-definition.json
```

- Edit your `openapi-definition.json` file in your favorite text editor

> - Watch the live preview being updated each time your save your file!

Reviewer: ```suggestion
- Watch the live preview being updated each time you save your file.
```
